### PR TITLE
[DOR 38 & 39] Search results page styling updates

### PR DESCRIPTION
**Pull request by janehughes**
Ticket URLs: 

- https://national-archives.atlassian.net/jira/software/c/projects/DOR/boards/128?assignee=unassigned&assignee=70121%3Ac27fc817-0a00-4972-8745-62c84f4f8c65&selectedIssue=DOR-38
- https://national-archives.atlassian.net/jira/software/c/projects/DOR/boards/128?assignee=unassigned&assignee=70121%3Ac27fc817-0a00-4972-8745-62c84f4f8c65&selectedIssue=DOR-39

## About these changes

Updates the 'Catalogue' search results page to be closer in style to those in the figma prototypes.

https://www.figma.com/file/pcIxOpwvJ1YRYbt9lbeKUG/TNA-Etna---Design-system?type=design&node-id=9449-49323&mode=design&t=9oLhhSJdB2ppXVjC-0

image

There are likely going to be many more refinements and tweaks needed, especially as this is currently a bit of a halfway house between the existing search and what will inevitably use the TNA front-end. We might want to consider updating some of the styles later, for consistency's sake.

## How to check these changes

Check figma designs and compare with updated FE.

Use http://localhost:8000/search/catalogue/ for testing locally.

Note that we don't yet have the fully functioning search, or the thumbnails pulling in for records. Hopefully this will be available to update soon.

## Before assigning to reviewer, please make sure you have

- [ ] Checked things thoroughly before handing over to reviewer
- [ ] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [ ] Ensured that PR includes only commits relevant to the ticket
- [ ] Waited for all CI jobs to pass before requesting a review
- [ ] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/search/search.html [hunk 1]
```diff
@@ -9,15 +11,14 @@
 {% block content %}
 
 {% include './blocks/search_landing_hero.html' %}
-
-    <div class="search-landing-buckets">
-        <div class="tna-container">
+    {% comment %}<div class="search-landing-buckets">
```
Comment (tm-kn) [inline templates/search/search.html:14]: **Suggestion:** Could think about explaining why this is being commented out so it does not make it into any real code.

**Comment by tm-kn** [comment]
Previously there was a disabled "previous" button on the search pagination. @janehughes now has removed it if there's no previous page available. That's what is breaking the CI tests now.

@AJ Do you know if we'd prefer to have a disabled previous button or is it okay if it's completely invisible? Looking at the GOV.UK design system [they recommend not to show the button if there's no previous page](https://design-system.service.gov.uk/components/pagination/#first-and-last-pages) so what @janehughes implemented would align with that.

**Comment by tm-kn** [comment]
Previously there was a disabled "previous" button on the search pagination. @janehughes now has removed it if there's no previous page available. That's what is breaking the CI tests now.

@ahosgood Do you know if we'd prefer to have a disabled previous button or is it okay if it's completely invisible? Looking at the GOV.UK design system [they recommend not to show the button if there's no previous page](https://design-system.service.gov.uk/components/pagination/#first-and-last-pages) so what @janehughes implemented would align with that.

**Comment by ahosgood** [comment]
@tm-kn I would prefer no button at all. We shouldn't be using disabled buttons.

I had a PR to update the pagination as part of a TNA Frontend update but there were other issues so I haven't touched it in ages.

Remove the button 👍🏼

File: scripts/src/modules/search/search-sort-by.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default function () {
+    const sortByForm = document.querySelector("[data-sort-search]");
+    const sortBySelector = sortByForm.querySelector("#id_sort_by_desktop");
+
+    sortBySelector.addEventListener("change", function () {
+        console.log("clicked select");
+        sortByForm.submit();
```
Comment (tm-kn) [inline scripts/src/modules/search/search-sort-by.js:7]: **Issue:** This assumes that `sortByForm` is not null.

File: scripts/src/modules/search/search-sort-by.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export default function () {
+    const sortByForm = document.querySelector("[data-sort-search]");
+    const sortBySelector = sortByForm.querySelector("#id_sort_by_desktop");
+
+    sortBySelector.addEventListener("change", function () {
```
Comment (tm-kn) [inline scripts/src/modules/search/search-sort-by.js:5]: **Issue:** This assumes that `sortBySelector` is not null.

File: scripts/src/modules/search/search-sort-by.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+        console.log("clicked select");
```
Comment (tm-kn) [inline scripts/src/modules/search/search-sort-by.js:6]: **Question:** I'm imagining this console log is not needed.

File: scripts/src/modules/search/search-sort-by.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export default function () {
+    const sortByForm = document.querySelector("[data-sort-search]");
+    const sortBySelector = sortByForm.querySelector("#id_sort_by_desktop");
+
```
Comment (ahosgood) [inline scripts/src/modules/search/search-sort-by.js:4]:
```suggestion
if(!sortByForm || !sortBySelector) {
  return
}
```

**Review by ahosgood — APPROVED** [review]
This looks good enough to go into the `experiment/image-viewer` branch for now. Will give a more thorough review once/if it's ready for `develop`.

I left a suggestion that should help with the concerns @tm-kn raised.

File: scripts/src/modules/search/search-sort-by.js (hunk 4 above)
Comment (janehughes) [reply scripts/src/modules/search/search-sort-by.js:6]: You're correct - thanks!

File: scripts/src/modules/search/search-sort-by.js (hunk 5 above)
Comment (janehughes) [reply scripts/src/modules/search/search-sort-by.js:4]: 👍  thanks :)

**Pull request merged**